torch/distributed/algorithms/fsdp_comm_hooks/allreduce_hook.py

```diff
@@ -0,0 +1,49 @@
+import torch
```

Comment:
Does it make sense if we start as a private/prototype feature (i.e., `_fsdp_comm_hooks`) first? Or is this planned to be a beta feature in the v1.13 release?

Reply:
That is not planned, thanks for pointing this out. Moved to a private folder `_comm_hooks`